File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        fPtr = head
        sPtr = head
        counter = 0
        result = []
        while(fPtr):
            fPtr = fPtr.next
            counter += 1
        for i in range(0, counter//2 ):
            sPtr = sPtr.next
        return sPtr
            
            
            
        

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode slowPtr = head, fastPtr = head;
        int counter = 0;
        
        while(fastPtr  != null){
            counter++;
            fastPtr = fastPtr.next;
        }
        counter = counter/2;
        while(counter > 0){
            counter--;
            slowPtr = slowPtr.next;
        }
        return slowPtr; 
        
    }
}